test/core/security/credentials_test.cc

```diff
@@ -3907,6 +3920,349 @@ TEST(
+      "\\\"urn:ietf:params:oauth:token-type:saml2\\\",\\\"saml_response\\\":"
+      "\\\"test_subject_token\\\",\\\"expiration_time\\\":%d}\" > %s",
+      expiration_time, output_file_filename);
+  return gpr_strdup(executable_file_contents.c_str());
+}
+
+char* get_pluggable_auth_executable_file_contents(int32_t expiration_time) {
+  std::string executable_file_contents = absl::StrFormat(
+      "#!/bin/bash\n\necho \"{\\\"version\\\": 1,\\\"success\\\": "
+      "\"true\",\\\"token_type\\\": "
+      "\\\"urn:ietf:params:oauth:token-type:saml2\\\",\\\"saml_response\\\":"
+      "\\\"test_subject_token\\\",\\\"expiration_time\\\":%d}\"",
+      expiration_time);
+  return gpr_strdup(executable_file_contents.c_str());
+}
+
+Json get_valid_pluggable_auth_credential_source(const char* filename) {
+  auto credential_source = JsonParse(
+      absl::StrFormat("{\"executable\":{\"command\":\"%s\"}}", filename));
+  GPR_ASSERT(credential_source.ok());
+  return *credential_source;
+}
+
+Json get_valid_pluggable_auth_credential_source_with_output_file(
+    const char* executable_filename, const char* output_file_filename) {
+  auto credential_source = JsonParse(absl::StrFormat(
+      "{\"executable\":{\"command\":\"%s\", \"output_file\":\"%s\"}}",
+      executable_filename, output_file_filename));
+  GPR_ASSERT(credential_source.ok());
+  return *credential_source;
+}
+
+TEST(CredentialsTest, TestPluggableAuthExecutableSuccess) {
+  ExecCtx exec_ctx;
+  char* executable_file_contents = get_pluggable_auth_executable_file_contents(
+      gpr_time_to_millis(gpr_inf_future(GPR_CLOCK_REALTIME)));
+  char* filename =
+      write_file(pluggable_auth_file_path_prefix, executable_file_contents);
+  auto credential_source = get_valid_pluggable_auth_credential_source(filename);
+  TestExternalAccountCredentials::ServiceAccountImpersonation
+      service_account_impersonation;
+  service_account_impersonation.token_lifetime_seconds = 3600;
+  ExternalAccountCredentials::Options options = {
+      "external_account",                 // type;
+      "audience",                         // audience;
+      "subject_token_type",               // subject_token_type;
+      "",                                 // service_account_impersonation_url;
+      service_account_impersonation,      // service_account_impersonation;
+      "https://foo.com:5555/token",       // token_url;
+      "https://foo.com:5555/token_info",  // token_info_url;
+      credential_source,                  // credential_source;
+      "quota_project_id",                 // quota_project_id;
+      "client_id",                        // client_id;
+      "client_secret",                    // client_secret;
+      "",                                 // workforce_pool_user_project;
+  };
+  grpc_error_handle error;
+  auto creds =
+      PluggableAuthExternalAccountCredentials::Create(options, {}, &error);
+  SetEnv("GOOGLE_EXTERNAL_ACCOUNT_ALLOW_EXECUTABLES", "1");
```

Comment:
Nit: Can you read what is currently set for the env variable and set it back to that rather than setting and then unsetting? Don't think it would come up but possible that someone runs the tests on a machine that already has it set to true

Reply:
Looks like the tests run in an isolated environment and don't affect the env variables of the machine. So, is it okay to leave this as it is?

Reply:
Is that documented somewhere? How did you figure that out